### PR TITLE
Move kubeproxy configuration to code

### PR DESCRIPTION
**Pull request by justinsb**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2206)

### CODE REVIEW

File: nodeup/pkg/model/kubeproxy.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	flags, err := flagbuilder.BuildFlags(c)
+	if err != nil {
+		return nil, fmt.Errorf("error building kubeproxy flags: %v", err)
+	}
+
+	image := c.Image
+	cmd := "echo -998 > /proc/$$$/oom_score_adj && kube-proxy --kubeconfig=/var/lib/kube-proxy/kubeconfig --resource-container = \"\" "
+	cmd += flags
+	// TODO: tee or similar so we get logs in kubectl logs
+	cmd += " 1>>/var/log/kube-proxy.log 2>&1"
```
Comment (chrislovecnm) [inline nodeup/pkg/model/kubeproxy.go]: Can we do this now?  Sucks not having these in docker.

File: nodeup/pkg/model/kubeproxy.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+	command := []string{
+		"/bin/sh", "-c",
+		cmd,
+	}
+
+	container := &v1.Container{
+		Name:            "kube-proxy",
+		Image:           image,
+		ImagePullPolicy: v1.PullIfNotPresent,
```
Comment (chrislovecnm) [inline nodeup/pkg/model/kubeproxy.go]: Can we make this Always?  Or should we?

File: nodeup/pkg/model/kubeproxy.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	{
+		kubeconfig, err := b.BuildPKIKubeconfig("kube-proxy")
+		if err != nil {
+			return err
+		}
+		t := &nodetasks.File{
+			Path:     "/var/lib/kube-proxy/kubeconfig",
+			Contents: fi.NewStringResource(kubeconfig),
+			Type:     nodetasks.FileType_File,
+			Mode:     s("0400"),
```
Comment (kris-nova) [inline nodeup/pkg/model/kubeproxy.go]: I think this is my first time ever seeing 400 in the wild.. Nicely done <3

File: nodeup/pkg/model/kubeproxy.go [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+		return nil, fmt.Errorf("error building kubeproxy flags: %v", err)
+	}
+
+	image := c.Image
+	cmd := "echo -998 > /proc/$$$/oom_score_adj && kube-proxy --kubeconfig=/var/lib/kube-proxy/kubeconfig --resource-container = \"\" "
+	cmd += flags
+	// TODO: tee or similar so we get logs in kubectl logs
+	cmd += " 1>>/var/log/kube-proxy.log 2>&1"
+
+	command := []string{
```
Comment (kris-nova) [inline nodeup/pkg/model/kubeproxy.go]: Can/should we log the full command here after we build it?

File: nodeup/pkg/model/kubeproxy.go (hunk 2 above)
Comment (justinsb) [inline nodeup/pkg/model/kubeproxy.go]: The version should identify the image uniquely

File: nodeup/pkg/model/kubeproxy.go (hunk 1 above)
Comment (justinsb) [inline nodeup/pkg/model/kubeproxy.go]: I can check if we have `tee` in the image, but otherwise I don't know of an easy way

**Pull request merged**